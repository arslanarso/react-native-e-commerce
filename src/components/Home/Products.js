/* eslint-disable react-hooks/rules-of-hooks */
import {
  View,
  Text,
  Image,
  TouchableOpacity,
  StyleSheet,
  Dimensions,
  Pressable,
  Alert,
} from 'react-native';
import FastImage from 'react-native-fast-image';
import React, {useState} from 'react';
import {useNavigation} from '@react-navigation/native';
import {useDispatch, useSelector} from 'react-redux';
import {
  addToCart,
  removeFromCart,
  incrementQuantity,
  decrementQuantity,
} from '../../../Redux/CartReducer';

const Products = props => {
  const navigation = useNavigation();
  const [_categories, setCategories] = useState('');
  const cart = useSelector(state => state.cart.cart);

  const dispatch = useDispatch();

  const categoriesList = _categories;
  const item = props.item;

  const addItemToCart = item => {
    dispatch(addToCart(item));
  };
  const removeItemFromCart = item => {
    dispatch(removeFromCart(item));
  };

  const AddToCartButton = () => {
    return (
      <>
        {cart.some(value => value.id == item.id) ? (
          <Pressable
            style={styles.removeButton}
            onPress={() => {
              Alert.alert('Başarılı', 'Ürün Başarıyla Sepetten Çıkartıldı', [
                {text: 'Tamam', onPress: () => {}},
              ]);
              removeItemFromCart(item);
            }}>
            <Text
              style={{
                color: '#f70c02',
                fontWeight: 'bold',
              }}>
              SİL
            </Text>
          </Pressable>
        ) : (
          <Pressable
            style={styles.button}
            onPress={() => {
              Alert.alert('Başarılı', 'Ürün Başarıyla Sepete Eklendi', [
                {text: 'Tamam', onPress: () => {}},
              ]);
              addItemToCart(item);
            }}>
            <Text
              style={{
                color: 'white',
                fontWeight: 'bold',
              }}>
              Sepete Ekle
            </Text>
          </Pressable>
        )}
      </>
    );
  };

  return (
    <TouchableOpacity
      onPress={() =>
        navigation.navigate('Ürün Detay', {
          item: item,
        })
      }
      style={styles.root}>
      <View>
        <FastImage
          style={[styles.image]}
          source={{
            uri: item.image,
            priority: FastImage.priority.normal,
          }}
          resizeMode={FastImage.resizeMode.cover}
        />
      </View>
      <View
        style={{
          justifyContent: 'center',
          margin: 10,
          width: 150,
        }}>
        <Text style={styles.title}>{item.price} ₺</Text>
        <Text numberOfLines={1} style={{marginVertical: 5, fontSize: 16}}>
          {item.name}
        </Text>
        <View style={{justifyContent: 'flex-end'}}>
          <AddToCartButton />
        </View>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  root: {
    borderRadius: 5,
    backgroundColor: 'white',
    marginVertical: 8,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,

    elevation: 5,
  },
  button: {
    backgroundColor: 'blue',
    alignItems: 'center',
    justifyContent: 'center',
    padding: 10,
    borderRadius: 8,
    marginTop: 5,
  },
  removeButton: {
    backgroundColor: '#fc8f8b',
    alignItems: 'center',
    justifyContent: 'center',
    padding: 10,
    borderRadius: 8,
    marginTop: 5,
  },

  image: {
    height: 200,
    borderTopLeftRadius: 5,
    borderTopRightRadius: 5,
    width: Dimensions.get('screen').width / 2.2,
    resizeMode: 'cover',
  },

  title: {
    fontSize: 16,
    marginTop: 5,
    color: 'blue',
    fontWeight: '700',
  },

  price: {
    fontSize: 18,
    fontWeight: 'bold',
  },
});

export default Products;
