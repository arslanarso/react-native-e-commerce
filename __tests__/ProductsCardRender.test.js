import React from 'react';
import renderer from 'react-test-renderer';
import Products from '../src/components/Home/Products';
import {NavigationContainer} from '@react-navigation/native';
import {Provider} from 'react-redux';
import store from '../Redux/store';

it('renders correctly', () => {
  const item = {
    id: 1,
    name: 'Test Product',
    price: 19.99,
    image: 'https://example.com/test.jpg',
    quantity: 1,
  };
  const tree = renderer
    .create(
      <NavigationContainer>
        <Provider store={store}>
          <Products item={item} />
        </Provider>
      </NavigationContainer>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
