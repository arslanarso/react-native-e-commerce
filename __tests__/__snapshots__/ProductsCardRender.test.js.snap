// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "white",
      "borderRadius": 5,
      "elevation": 5,
      "marginVertical": 8,
      "opacity": 1,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
    }
  }
>
  <View>
    <View
      style={
        [
          {
            "overflow": "hidden",
          },
          [
            {
              "borderTopLeftRadius": 5,
              "borderTopRightRadius": 5,
              "height": 200,
              "resizeMode": "cover",
              "width": 340.9090909090909,
            },
          ],
        ]
      }
    >
      <FastImageView
        defaultSource={null}
        resizeMode="cover"
        source={
          {
            "priority": "normal",
            "uri": "https://example.com/test.jpg",
          }
        }
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "position": "absolute",
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../src/assets/icons/emptyStar.png",
        }
      }
      style={
        {
          "height": 40,
          "width": 40,
        }
      }
    />
  </View>
  <View
    style={
      {
        "justifyContent": "center",
        "margin": 10,
        "width": 150,
      }
    }
  >
    <Text
      style={
        {
          "color": "blue",
          "fontSize": 16,
          "fontWeight": "700",
          "marginTop": 5,
        }
      }
    >
      19.99
       â‚º
    </Text>
    <Text
      numberOfLines={1}
      style={
        {
          "fontSize": 16,
          "marginVertical": 5,
        }
      }
    >
      Test Product
    </Text>
    <View
      style={
        {
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "blue",
            "borderRadius": 8,
            "justifyContent": "center",
            "marginTop": 5,
            "padding": 10,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontWeight": "bold",
            }
          }
        >
          Sepete Ekle
        </Text>
      </View>
    </View>
  </View>
</View>
`;
